examples/with-metadata.rs

```diff
@@ -0,0 +1,144 @@
+//! A single threaded executor that uses shortest-job-first scheduling.
+
+use std::cell::RefCell;
+use std::collections::BinaryHeap;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::thread;
+use std::time::{Duration, Instant};
+use std::{cell::Cell, future::Future};
+
+use async_task::{Builder, Runnable, Task};
+use pin_project_lite::pin_project;
+use smol::{channel, future};
+
+struct ByDuration(Runnable<DurationMetadata>);
```

Comment:
In theory I should store this as a (Duration, Runnable) pair, to avoid any Cell-affecting-Ord shenanigans. Given the only place that modifies this is the `MeasureRuntime::poll` method, which is only called when the runnable is not in the heap, this isn't actually a problem.

Reply:
I like this better.